modules/sc-mesh-secure-deployment/src/2_0/features/cbma/auth/authServer.py

```diff
@@ -0,0 +1,136 @@
+                self.mua.connected_peers_status[client_mac] = ["ongoing",0]  # Update status as ongoing, num of failed attempts = 0
+        else:
+            with self.mua.connected_peers_status_lock:
+                self.mua.connected_peers_status[client_mac][0] = "ongoing"  # Update status as ongoing, num of failed attempts = same as before
+        self.authenticate_client(client_connection, client_address, client_mac)
+
+    def authenticate_client(self, client_connection, client_address, client_mac):
+        secure_client_socket = self.context.wrap_socket(client_connection, server_side=True)
+        try:
+            client_cert = secure_client_socket.getpeercert(binary_form=True)
+            if not client_cert:
+                logger.error(f"Unable to get the certificate from the client {client_address[0]}", exc_info=True)
+                raise CertificateNoPresentError("Unable to get the certificate from the client")
+
+            auth = verify_cert(client_cert, self.ca, client_address[0], self.interface, logger)
+            with self.client_auth_results_lock:
+                self.client_auth_results[client_address[0]] = auth
+            if auth:
+                with self.active_sockets_lock:
+                    self.active_sockets[client_address[0]] = secure_client_socket
+                self.mua.auth_pass(secure_client_socket=secure_client_socket, client_mac=client_mac)
+            else:
+                # Handle the case when authentication fails, maybe send an error message
+                self.mua.auth_fail(client_mac=client_mac)
+                secure_client_socket.send(b"Authentication failed.")
+        except Exception as e:
+            logger.error(f"An error occurred while handling the client {client_address[0]}.", exc_info=True)
+            self.mua.auth_fail(client_mac=client_mac)
+        # finally:
+        #     secure_client_socket.close()
+
+    def get_secure_socket(self, client_address):
+        with self.active_sockets_lock:
+            return self.active_sockets.get(client_address)
+
+    def get_client_auth_result(self, client_address):
+        with self.client_auth_results_lock:
+            return self.client_auth_results.get(client_address, None)
+
+    def start_server(self):
+        if is_ipv4(self.ipAddress):
+            self.serverSocket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+            self.serverSocket.bind((self.ipAddress, self.port))
+        elif is_ipv6(self.ipAddress):
+            self.serverSocket = socket.socket(socket.AF_INET6, socket.SOCK_STREAM)
+            scope_id = socket.if_nametoindex(self.interface)
+            self.serverSocket.bind((self.ipAddress, int(self.port), 0, scope_id))
+        else:
+            raise ValueError("Invalid IP address")
+
+        self.serverSocket.listen()
+        self.serverSocket.settimeout(99999)  # maybe we can remove timeout since server needs to be listening throughout
```

Comment:
[Python experts please confirm]

If I hastily understand the documentation correctly, settimeout() is for specifying how many seconds we want to wait on each blocking socket call so 99999 is way too long as it would enable a client to keep us stuck for more than a day which is a denial of service in the main thread but also quite inconvenient in client threads...

Please change to 30s which is still a bit overkill but for safety in case an Upper MACsec client is dozens of hops away, and having something fairly small also gives us an opportunity to cleanup regularly in a "garbage collection" fashion when there is nothing else to do 🙂